src/loki/reach.cc

```diff
@@ -41,38 +49,49 @@ directed_reach SimpleReach(const DirectedEdge* edge,
     if (node_filter(node))
       return;
     // otherwise we enqueue it
-    queue.insert(node_id);
+    queue_.insert(node_id);
     // and we enqueue it on the other levels
     for (const auto& transition : tile->GetNodeTransitions(node))
-      queue.insert(transition.endnode());
+      queue_.insert(transition.endnode());
     // and we remember how many duplicates we enqueued
-    transitions += node->transition_count();
+    transitions_ += node->transition_count();
   };
 
+  // TODO: here we stay extra conservative by avoiding starting on a simple restriction because we
+  // TODO: dont have predecessor information in the simple reach expansion so we bail 1 edge earlier
+  // NOTE: we can expand from the end of a complex restriction because it cant possibly be on our path
+  // and we can expand from the start of a complex restriction because only the end would mark a
+  // potential stopping point (maybe a path followed the restriction)
```

Comment:
this is important, its how the conservative reach works. for the forward expansion we cannot start with a simple restriction because we have no predecessor info to let us know if a path would have activated the restriction.